src/app/shared/navigation-focus/navigation-focus.service.ts

```diff
@@ -0,0 +1,43 @@
+import {ElementRef, Injectable} from '@angular/core';
+import {NavigationEnd, Router} from '@angular/router';
+import {filter, skip, take} from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NavigationFocusService {
+  private navigationFocusRequests: ElementRef[] = [];
+  private skipLinkFocusRequests: ElementRef[] = [];
+
+  constructor(private router: Router) {
+    this.router.events
+      .pipe(filter(event => event instanceof NavigationEnd),
+        take(1))
+      .subscribe(() => {
+        setTimeout(() =>
+          this.skipLinkFocusRequests[this.skipLinkFocusRequests.length - 1].nativeElement.id =
+            'main-content', 100);
+      });
+
+    this.router.events
+      .pipe(filter(event => event instanceof NavigationEnd),
+        skip(1))
+      .subscribe(() => {
+        if (!window.location.hash) {
```

Comment:
is there still code somewhere that handles the fragment links or did we remove that?

Reply:
I think that was removed

Reply:
Can you get the fragment from the `Router` state instead or at least from some injectable service? Otherwise you'll have trouble testing this.